gf2_192/src/gf2_192poly.rs

```diff
@@ -0,0 +1,242 @@
+            return Err(Gf2_192PolyError::InterpolatePointsAndValuesLengthDiffer);
+        }
+
+        let result_degree = values.len();
+
+        let mut result = Gf2_192Poly::make_constant(result_degree, 0);
+        let mut vanishing_poly = Gf2_192Poly::make_constant(result_degree, 1);
+
+        for i in 0..points.len() {
+            let mut t = result.evaluate(points[i]);
+            let mut s = vanishing_poly.evaluate(points[i]);
+
+            // need to find r such that currentValue+r*valueOfVanishingPoly = values[i]
+            t = t + values[i];
+            s = Gf2_192::invert(s);
+            t = t * s;
+
+            result.add_monic_times_constant(vanishing_poly.clone(), t);
+
+            vanishing_poly.multiply_by_linear_binomial(points[i]);
+        }
+
+        // Last point is at 0
+        let mut t = result.coefficients[0]; // evaluating at 0 is easy
+        let mut s = vanishing_poly.coefficients[0]; // evaluating at 0 is easy
+
+        // need to find r such that currentValue+r*valueOfVanishingPoly = valueAt0]
+        t = t + value_at_zero;
+        s = Gf2_192::invert(s);
+        t = t * s;
+        result.add_monic_times_constant(vanishing_poly, t);
+
+        Ok(result)
+    }
+
+    /// Evaluates polynomial at the given point `x`, where `x` represents the last byte of a field
+    /// element (all other bits are assumed to be 0)
+    pub fn evaluate(&self, x: i8) -> Gf2_192 {
+        let mut res = self.coefficients[self.degree];
+        if self.degree > 0 {
+            for d in (0..=(self.degree - 1)).rev() {
+                res = Gf2_192::mul_by_i8(res, x);
+                res = res + self.coefficients[d];
+            }
+        }
+        res
+    }
+
+    /// Returns Vec consisting of the concatenation of all the coefficients of the polynomial.
+    ///  - Degree-zero coefficient is located at index 0
+    ///  - Each coefficient takes 24 bytes for a total of `(self.degree+1)*24` bytes
+    pub fn to_i8_vec(&self) -> Vec<i8> {
+        let mut res: Vec<_> = std::iter::repeat(0).take((self.degree + 1) * 24).collect();
+        for i in 0..self.degree {
+            #[allow(clippy::unwrap_used)]
+            self.coefficients[i].to_i8_slice(&mut res, i * 24).unwrap();
+        }
+        res
+    }
+
+    /// Adds r*p to `self`. Assumes:
+    ///  - p is monic
+    ///  - self.coefficients.len() > p.degree
+    ///  - p.degree == self.degree + 1 or (self == 0 and p == 1)
+    fn add_monic_times_constant(&mut self, p: Gf2_192Poly, r: Gf2_192) {
+        let mut _t = Gf2_192::new();
+        for i in 0..p.degree {
+            _t = p.coefficients[i] * r;
+            self.coefficients[i] = self.coefficients[i] + _t;
+        }
+        self.degree = p.degree;
+        self.coefficients[self.degree] = r;
+    }
+
+    /// Multiply `self` by `x + r`. Assumes that `self` is monic (i.e.
+    /// `self.coefficients[self.degree] == 1`)
+    fn multiply_by_linear_binomial(&mut self, r: i8) {
```

Comment:
This method is named `monicTimesMonomial` by Reyzin, but I checked the definitions and I'm sure it's not a monomial but rather a linear binomial.